src/cargo/ops/cargo_compile.rs

```diff
@@ -163,6 +167,23 @@ pub fn compile_pkg(package: &Package, options: &CompileOptions)
     let to_build = packages.iter().find(|p| p.package_id() == pkgid).unwrap();
     let targets = try!(generate_targets(to_build, mode, filter, release));
 
+    let target_with_args = match target_rustc_args {
+        &Some(args) => {
+            if targets.len() > 1 {
+                return Err(human("extra arguments to `rustc` can only be \
+                                  invoked for one target"))
```

Comment:
Yeah I agree that it would be nice to improve this error message a bit. It would be nice for it to mention the filtering utilities (e.g. `--lib` and `--bin`) and then also perhaps provide a suggestion along the lines of "consider passing `--lib`" (or something like that)